boa_gc/src/lib.rs

```diff
@@ -1,6 +1,375 @@
 //! Garbage collector for the Boa JavaScript engine.
 
-pub use gc::{
-    custom_trace, finalizer_safe, force_collect, unsafe_empty_trace, Finalize, Gc, GcCell as Cell,
-    GcCellRef as Ref, GcCellRefMut as RefMut, Trace,
-};
+#![warn(
+    clippy::perf,
+    clippy::single_match_else,
+    clippy::dbg_macro,
+    clippy::doc_markdown,
+    clippy::wildcard_imports,
+    clippy::struct_excessive_bools,
+    clippy::doc_markdown,
+    clippy::semicolon_if_nothing_returned,
+    clippy::pedantic
+)]
+#![deny(
+    clippy::all,
+    clippy::cast_lossless,
+    clippy::redundant_closure_for_method_calls,
+    clippy::unnested_or_patterns,
```

Comment:
```suggestion
    clippy::redundant_closure_for_method_calls,
    clippy::use_self,
    clippy::unnested_or_patterns,
```

Let's add the `use_self` lint to make sure we catch the places that @Razican mentioned in his comment.